### ISSUE TITLE
`sym(true) + 1` leaves a mess

### ISSUE THREAD

**Issue by cbm755** [cbm755/octsympy#118]
````
AttributeError: 'BooleanTrue' object has no attribute 'as_coeff_Mul'
````
Need a `sympref reset` to recover.

This is basically issue #79.

**Comment by latot** [comment]
mm, this is weird check this, two differents outputs:
```
>> sym(true)+1
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 12, in octoutput_drv
  File "<stdin>", line 52, in octoutput
  File "<stdin>", line 33, in octoutput
  File "/usr/lib64/python2.7/site-packages/sympy/printing/pretty/pretty.py", line 2030, in pretty
    return pp.doprint(expr)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/pretty/pretty.py", line 58, in doprint
    return self._print(expr).render(**self._settings)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/printer.py", line 257, in _print
    return getattr(self, printmethod)(expr, *args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/pretty/pretty.py", line 1234, in _print_Add
    terms = self._as_ordered_terms(expr, order=order)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/printer.py", line 271, in _as_ordered_terms
    return expr.as_ordered_terms(order=order)
  File "/usr/lib64/python2.7/site-packages/sympy/core/expr.py", line 889, in as_ordered_terms
    terms, gens = self.as_terms()
  File "/usr/lib64/python2.7/site-packages/sympy/core/expr.py", line 918, in as_terms
    coeff, _term = term.as_coeff_Mul()
AttributeError: 'BooleanTrue' object has no attribute 'as_coeff_Mul'
error: sq_string cannot be indexed with {
error: called from
    python_cmd at line 144 column 3
    axplusy at line 40 column 5
    plus at line 39 column 5
error: evaluating argument list element number 1
error: called from
    python_cmd at line 144 column 3
    axplusy at line 40 column 5
    plus at line 39 column 5
```
second:
```
>> python_cmd('return S.true + 1')
error: TypeError: unsupported operand type(s) for +: 'BooleanTrue' and 'int'
error: called from
    python_cmd at line 145 column 5
```
if its an upstream error i think the output should be equal..

**Comment by cbm755** [comment]
ok, noted in upstream bug.

**Comment by latot** [comment]
mm, i only get this running via sum in octave:
```
>> python_cmd('return S.true+1')
error: Python raised exception, maybe at line 1 of code block (assuming popen2)
TypeError: unsupported operand type(s) for +: 'BooleanTrue' and 'int'
error: called from
    python_cmd at line 145 column 5
>> python_cmd('return 1+S.true')
error: Python raised exception, maybe at line 1 of code block (assuming popen2)
TypeError: unsupported operand type(s) for +: 'int' and 'BooleanTrue'
error: called from
    python_cmd at line 145 column 5
```

**Comment by latot** [comment]
:O with 1+sum(true) we can know the error:
```
>> 1+sym(true)
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 12, in octoutput_drv
  File "<stdin>", line 50, in octoutput
  File "<stdin>", line 30, in octoutput
  File "/usr/lib64/python2.7/site-packages/sympy/core/basic.py", line 397, in __str__
    return sstr(self, order=None)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/str.py", line 748, in sstr
    s = p.doprint(expr)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/printer.py", line 233, in doprint
    return self._str(self._print(expr))
  File "/usr/lib64/python2.7/site-packages/sympy/printing/printer.py", line 257, in _print
    return getattr(self, printmethod)(expr, *args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/str.py", line 51, in _print_Add
    terms = self._as_ordered_terms(expr, order=order)
  File "/usr/lib64/python2.7/site-packages/sympy/printing/printer.py", line 271, in _as_ordered_terms
    return expr.as_ordered_terms(order=order)
  File "/usr/lib64/python2.7/site-packages/sympy/core/expr.py", line 889, in as_ordered_terms
    terms, gens = self.as_terms()
  File "/usr/lib64/python2.7/site-packages/sympy/core/expr.py", line 918, in as_terms
    coeff, _term = term.as_coeff_Mul()
AttributeError: 'BooleanTrue' object has no attribute 'as_coeff_Mul'
ans = PYTHON: Error in var export
```

**Comment by latot** [comment]
finally:
this works:
```
>> python_cmd('return _ins[0]+_ins[1]', 1, sym(true))
```
this don't works:
```
>> python_cmd('return _ins[0]+_ins[1]', sym(1), sym(true))
```

**Comment by latot** [comment]
mm, i think here are two problems, first this:
```
AttributeError: 'BooleanTrue' object has no attribute 'as_coeff_Mul'
```
and second "leaves a mess", with the actual code of octsympy the error should be handled, but it doesn't happend, my last comments was about that, wy some lines are handled and others not, sending a bad error and braking octsympy/octave.

**Comment by cbm755** [comment]
Ah!  Thanks I had forgotten about that.  I filed #346 so I don't miss it again: its much more important that how SymPy deals with Boolean addition.

**Comment by cbm755** [comment]
ok, the octsympy specific bits of this are fixed by #357, with tests in place.

upstream bits are upstream problem.
